packages/core/cli/__tests__/domains/git-repository-url/getGitRepositoryURL.spec.ts

```diff
@@ -0,0 +1,138 @@
+import git from 'isomorphic-git';
+import { t, TFunction } from 'i18next';
+import { stdin, MockSTDIN } from 'mock-stdin';
+import { getGitRepositoryURL, validateGitRepositoryURL } from '../../../src/domains/git-repository-url';
+
+jest.mock('i18next');
+jest.mock('../../../src/domains/git-repository-url/validateGitRepositoryURL');
+
+const ENTER_KEY = '\x0D';
+const BACKSPACE_KEY = '\x08';
+
+type MockValidate = jest.MockedFunction<typeof validateGitRepositoryURL>;
+
+const wait = (time: number): Promise<void> => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, time);
+  });
+};
```

Comment:
This can be useful applied in other contexts, why not extract it to a separate module?